temporian/core/event_set_ops.py

```diff
@@ -1613,6 +1742,222 @@ def begin(self: EventSetOrNode) -> EventSetOrNode:
 
         return begin(self)
 
+    def bitwise_and(self: EventSetOrNode, other: Any) -> EventSetOrNode:
+        """Computes the bitwise AND between an [`EventSet`][temporian.EventSet]
+        or a scalar value to `self` element-wise. The operation is only
+        supported for Events with features that are integers.
+
+        If an EventSet, each feature in `self` is bitwise ANDed to the feature
+        in `other` in the same position. `self` and `other` must have the same
+        sampling, index, number of features and dtype for the features in the
+        same positions.
+
+        If a scalar, `other` is bitwise ANDed with each item in each feature
+        in `self`.
+
+        Example with EventSet:
+            ```python
+            >>> a = tp.event_set(
+            ...     timestamps=[1, 2, 3],
+            ...     features={"f1": [2, 5, 7], "f2": [9, 11, 13]}
+            ... )
+            >>> b = tp.event_set(
+            ...     timestamps=[1, 2, 3],
+            ...     features={"f3": [3, 4, 8], "f4": [10, 12, 15]},
+            ...     same_sampling_as=a
+            ... )
+
+            >>> c = a.bitwise_and(b)
+            >>> c
+            indexes: []
+            features: [('bitwise_and_f1_f3', int64), ('bitwise_and_f2_f4', int64)]
+            events:
+                (3 events):
+                    timestamps: [1. 2. 3.]
+                    'bitwise_and_f1_f3': [2 4 0]
+                    'bitwise_and_f2_f4': [ 8  8 13]
+            ...
+
+            ```
+
+        Example with scalar value:
+            ```python
+            >>> a = tp.event_set(
+            ...     timestamps=[1, 2, 3],
+            ...     features={"f1": [2, 5, 7], "f2": [9, 11, 13]}
+            ... )
+
+            >>> b = a.bitwise_and(3)
+            >>> b
+            indexes: ...
+                    timestamps: [1. 2. 3.]
+                    'f1': [2 1 3]
+                    'f2': [1 3 1]
+            ...
+
+            ```
+
+        Args:
+            other: EventSet or scalar value.
+
+        Returns:
+            Result of the operation.
+        """
+        if isinstance(other, self.__class__):
+            from temporian.core.operators.binary import bitwise_and
+            return bitwise_and(input_1=self, input_2=other)
+
+        if isinstance(other, int):
+            from temporian.core.operators.scalar import bitwise_and_scalar
+            return bitwise_and_scalar(input=self, value=other)
+
+        self._raise_error("bitwise_and", other, "int")
+        assert False
+
+    def bitwise_or(self: EventSetOrNode, other: Any) -> EventSetOrNode:
+        """Computes the bitwise OR between an [`EventSet`][temporian.EventSet]
+        or a scalar value to `self` element-wise. The operation is only
+        supported for Events with features that are integers.
+
+        If an EventSet, each feature in `self` is bitwise ORed to the feature
```

Comment:
I'm not convinced about using "bitwise or" as a verb here, I would rephrase around

```suggestion
        If an EventSet, the bitwise or is calculated between each feature in `self` and the feature
```
If we do, `and` and `xor` need to be changed as well